cmd/torpedo/torpedo.go

```diff
@@ -336,14 +339,29 @@ func (t *torpedo) validateContext(ctx *scheduler.Context) error {
 	}
 
 	if err := t.validateVolumes(ctx); err != nil {
+		generateSupportBundle(t, ctx)
 		return err
 	}
 	if err = t.s.WaitForRunning(ctx); err != nil {
+		generateSupportBundle(t, ctx)
 		return err
 	}
 	return err
 }
 
+// generateSupportBundle gathers logs and any artifacts pertinent to the scheduler and dumps them in the defined location
+func generateSupportBundle(t *torpedo, ctx *scheduler.Context) error {
+	if t == nil || ctx == nil || t.s == nil {
+		return fmt.Errorf("Invalid context or torpedo object. Cannot generate support bundle")
+	}
+	out, err := t.s.GenerateSupportBundle(ctx)
+	err = ioutil.WriteFile(fmt.Sprintf("%s/supportbundle%v.log", t.logLoc, time.Now().Format(time.RFC3339)), []byte(out), 0644)
```

Comment:
Will do. I still want to leave the timestamp to disambiguate in case of multiple files, perhaps, at a future point or for merely a chronology reference.